File: MMORPGServer/Proto/CreateLuaProto/Goods_SearchEquipDetailProto.lua
--客户端发送查询装备详情消息
Goods_SearchEquipDetailProto = { ProtoCode = 16006, GoodsServerId = 0 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
Goods_SearchEquipDetailProto.__index = Goods_SearchEquipDetailProto;

function Goods_SearchEquipDetailProto.New()
    local self = { }; --初始化self
    setmetatable(self, Goods_SearchEquipDetailProto); --将self的元表设定为Class
    return self;
end


--发送协议
function Goods_SearchEquipDetailProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);

    ms:WriteInt(proto.GoodsServerId);

    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function Goods_SearchEquipDetailProto.GetProto(buffer)

    local proto = Goods_SearchEquipDetailProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);

    proto.GoodsServerId = ms:ReadInt();

    ms:Dispose();
    return proto;
end

File: MMORPGServer/GameServerApp/GameServerApp/Proto/Goods_SearchEquipDetailProto.cs
//===================================================
//作    者：边涯  http://www.u3dol.com  QQ群：87481002
//创建时间：2020-03-15 10:58:27
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary>
/// 客户端发送查询装备详情消息
/// </summary>
public struct Goods_SearchEquipDetailProto : IProto
{
    public ushort ProtoCode { get { return 16006; } }

    public int GoodsServerId; //物品服务器端编号

    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            ms.WriteInt(GoodsServerId);
            return ms.ToArray();
        }
    }

    public static Goods_SearchEquipDetailProto GetProto(byte[] buffer)
    {
        Goods_SearchEquipDetailProto proto = new Goods_SearchEquipDetailProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
            proto.GoodsServerId = ms.ReadInt();
        }
        return proto;
    }
}